core/typescript/src/lib/metadata.ts

```diff
@@ -17,6 +17,14 @@ export interface OptionalIconAware {
   readonly icon?: string;
 }
 
+/**
+ * Interface that offers an optional label field.
+ * This should be short and concise.
+ */
+export interface OptionalLabelAware {
+  readonly label?: string;
+}
+
```

Comment:
This is mostly for diagram tooling, from the latest discussion around how to build it where each component|service|relation type needs a label, i.e. name to show on a diagram, alongside description and icon which are already here.